src/main/java/com/google/devtools/build/lib/packages/StructImpl.java

```diff
@@ -299,7 +300,7 @@ private void printJson(Object value, StringBuilder sb, Location loc, String cont
           throw new EvalException(loc, errorMessage);
         }
         sb.append("\"");
-        sb.append(entry.getKey());
+        sb.append(StringEscapeUtils.escapeJava(String.valueOf(entry.getKey())));
```

Comment:
Don't use String.valueOf(x) here; it's equivalent to x.toString() except when x is null, which can't happen because this statement is dominated by the check on L291 (and even if it could, I would prefer to see a crash in that case than to silently emit incorrect JSON).

Use: (String) entry.getKey()

Reply:
Could you also apply escaping to the field names on L280? Field names should be valid identifiers, but this is not guaranteed, and we don't want to emit corrupt JSON in that case. 

At that point, there are three places that emit quote escape(s) quote; could you refactor jsonEscapeString to accept a StringBuilder and emit the quotes too? Thanks.

Reply:
Done.